src/webui/api/appcontroller.cpp

```diff
@@ -517,3 +517,18 @@ void AppController::defaultSavePathAction()
 {
     setResult(BitTorrent::Session::instance()->defaultSavePath());
 }
+
+void AppController::categoriesAction()
+{
+    QVariantHash categories;
+    const auto categoriesList = BitTorrent::Session::instance()->categories();
+    for (auto it = categoriesList.cbegin(); it != categoriesList.cend(); ++it) {
+        const auto key = it.key();
```

Comment:
explicit type please.

Reply:
The `auto` section of coding guidelines explicitly says iterators are a good application:

> Declarations for which one can gather enough information about the object interface (type) from its name or the usage pattern (an iterator or a loop variable are good examples of clear patterns) or the right part of the expression nicely fit here.

Reply:
I see you meant the key. This code was copied from `synccontroller` and knowing the explicit type of key here doesn't help us at all.